grpc-protocol/src/main/java/com/linecorp/armeria/client/grpc/protocol/UnaryGrpcClient.java

```diff
@@ -164,36 +165,40 @@ public HttpResponse execute(ClientRequestContext ctx, HttpRequest req) {
-                                   } catch (Exception e) {
-                                       throw new ArmeriaStatusException(StatusCodes.INTERNAL,
-                                                                        "Error executing request.");
+                                       try {
+                                           return unwrap().execute(ctx, HttpRequest.of(req.headers(), framed))
+                                                          .aggregateWithPooledObjects(ctx.eventLoop(),
+                                                                                      ctx.alloc());
+                                       } catch (Exception e) {
+                                           throw new ArmeriaStatusException(StatusCodes.INTERNAL,
+                                                                            "Error executing request.");
+                                       }
                                    }
                                })
                        .thenCompose(msg -> {
-                           if (!msg.status().equals(HttpStatus.OK) || msg.content().isEmpty()) {
-                               // Nothing to deframe.
-                               return CompletableFuture.completedFuture(msg.toHttpResponse());
-                           }
-
-                           final CompletableFuture<HttpResponse> responseFuture = new CompletableFuture<>();
+                           try (HttpData content = msg.content()) {
+                               if (msg.status() != HttpStatus.OK || content.isEmpty()) {
+                                   // Nothing to deframe.
+                                   return CompletableFuture.completedFuture(msg.toHttpResponse());
+                               }
 
-                           final ArmeriaMessageDeframer deframer =
-                                   new ArmeriaMessageDeframer(Integer.MAX_VALUE);
-                           msg.toHttpResponse().decode(deframer, ctx.alloc(),
-                                                       byteBufConverter(ctx.alloc(), isGrpcWebText))
-                              .subscribe(singleSubscriber(ctx, msg, serializationFormat, responseFuture),
-                                         ctx.eventLoop());
-                           return responseFuture;
+                               final CompletableFuture<HttpResponse> responseFuture = new CompletableFuture<>();
+                               final ArmeriaMessageDeframer deframer =
+                                       new ArmeriaMessageDeframer(Integer.MAX_VALUE);
+                               msg.toHttpResponse().decode(deframer, ctx.alloc(),
+                                                           byteBufConverter(ctx.alloc(), isGrpcWebText))
+                                  .subscribe(singleSubscriber(ctx, msg, serializationFormat, responseFuture),
```

Comment:
How about passing `isGrpcWeb` instead of `serializationFormat` though not related to this PR.

Reply:
serializationFormat.uriText() is used in the error message if trailers are malformed, similar to https://github.com/line/armeria/blob/fcd2d72a73218a3dda4c2c89f4908ec1ad12d810/grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaClientCall.java#L395-L397

Reply:
I cached isGrpcWeb in the member variable that's computed once on the subscriber instance creation, hope it addresses your concern, PTAL